src/components/Cluster/ClusterDetail/UpgradeClusterModalVersionChanger.tsx

```diff
@@ -0,0 +1,80 @@
+import ReleaseSelector from 'Cluster/NewCluster/ReleaseSelector/ReleaseSelector';
+import PropTypes from 'prop-types';
+import React from 'react';
+import BootstrapModal from 'react-bootstrap/lib/Modal';
+import Button from 'UI/Button';
+import StyledInput from 'UI/ClusterCreation/StyledInput';
+
+interface IUpgradeClusterModalVersionChangerProps
+  extends React.ComponentPropsWithoutRef<'div'> {
+  closeModal: () => void;
+  onSubmit: () => void;
+  onCancel: () => void;
+  onChangeRelease: (newRelease: string) => void;
+  releaseVersion: string;
+  isAdmin?: boolean;
+}
+
+const UpgradeClusterModalVersionChanger: React.FC<IUpgradeClusterModalVersionChangerProps> = ({
+  closeModal,
+  onSubmit,
+  onCancel,
+  onChangeRelease,
+  releaseVersion,
+  isAdmin,
+  ...rest
+}) => {
+  if (!isAdmin) {
+    onCancel();
+
+    return null;
+  }
+
+  return (
+    <div {...rest}>
```

Comment:
Good practice would be to change the modal title and body, instead of changing the whole modal, as we do with the cluster scale modal, but would've been too big of a change for this PR. So I just re-used the initial logic, for now